build-system/compile/pre-closure-babel.js

```diff
@@ -60,27 +45,62 @@ function getFilesToTransform() {
  * @return {!Promise}
  */
 function preClosureBabel() {
-  return through.obj((file, enc, next) => {
-    const cachedTransform = cachedTransforms[file.path];
-    if (cachedTransform) {
-      file.contents = Buffer.from(cachedTransform);
-    } else if (filesToTransform.includes(path.relative(ROOT_DIR, file.path))) {
-      const babelPlugins = conf.plugins({
-        isForTesting: !!argv.fortesting,
-        isEsmBuild: !!argv.esm,
-        isSinglePass: !!argv.single_pass,
-        isChecktypes: argv._.includes('check-types'),
-      });
-      const {code} = babel.transformFileSync(file.path, {
-        plugins: babelPlugins,
-        retainLines: true,
-        compact: false,
-      });
-      cachedTransforms[file.path] = code;
-      file.contents = Buffer.from(code);
+  const filesToTransform = getFilesToTransform();
+  const salt = [
+    fs.readFileSync(require.resolve('./build.conf.js')).toString('hex'),
+    fs.readFileSync('./babel.config.js').toString('hex'),
+    JSON.stringify(argv),
+  ].join(':');
+
+  const babelPlugins = conf.plugins({
+    isForTesting: !!argv.fortesting,
+    isEsmBuild: !!argv.esm,
+    isSinglePass: !!argv.single_pass,
+    isChecktypes: argv._.includes('check-types'),
+  });
+  const babel = gulpBabel({
+    plugins: babelPlugins,
+    retainLines: false,
+    compact: false,
+  });
+
+  function transform(file, enc, next) {
+    if (!filesToTransform.includes(file.relative)) {
+      return next(null, file);
     }
-    return next(null, file);
+
+    let data, err;
+    function onData(d) {
+      babel.off('error', onError);
+      data = d;
+    }
+    function onError(e) {
+      babel.off('data', onData);
+      err = e;
+    }
+    babel.once('data', onData);
+    babel.once('error', onError);
+    babel.write(file, enc, () => {
+      if (err) {
+        return next(err);
+      }
+
+      next(null, data);
+    });
+  }
```

Comment:
Curious: Where are these handlers implemented?

Reply:
Huh?

Reply:
It's Friday, and I wasn't reading carefully. Disregard!